python/tank/authentication/login_dialog.py

```diff
@@ -219,27 +282,47 @@ def _ok_pressed(self):
         if len(site) == 0:
             self._set_error_message(self.ui.message, "Please enter the address of the site to connect to.")
             return
-        if len(login) == 0:
-            self._set_error_message(self.ui.message, "Please enter your login name.")
-            return
-        if len(password) == 0:
-            self._set_error_message(self.ui.message, "Please enter your password.")
-            return
+        if not self._use_sso:
+            if len(login) == 0:
+                self._set_error_message(self.ui.message, "Please enter your login name.")
+                return
+            if len(password) == 0:
+                self._set_error_message(self.ui.message, "Please enter your password.")
+                return
 
         # if not protocol specified assume https
         if len(site.split("://")) == 1:
             site = "https://%s" % site
             self.ui.site.setText(site)
 
+        session_token = None
+        if self._use_sso:
+            login = ""
+            password = ""
+            res = self._saml2_sso.on_sso_login_attempt({
+                "host": site,
+                "cookies": self._cookies,
+                "product": "toolkit"
+            })
+            if res == QtGui.QDialog.Accepted:
+                session_token = self._saml2_sso._session.session_id
+                login = self._saml2_sso._session.user_id
+                self._cookies = self._saml2_sso._session.cookies
+            else:
+                error = self._saml2_sso.get_session_error()
+                if error:
+                    self._set_error_message(self.ui.message, error)
+                return
+
         try:
-            self._authenticate(self.ui.message, site, login, password)
+            self._authenticate(self.ui.message, site, login, password, session_token=session_token)
         except MissingTwoFactorAuthenticationFault:
             # We need a two factor authentication code, move to the next page.
             self.ui.stackedWidget.setCurrentWidget(self.ui._2fa_page)
         except Exception, e:
             self._set_error_message(self.ui.message, e)
 
-    def _authenticate(self, error_label, site, login, password, auth_code=None):
+    def _authenticate(self, error_label, site, login, password, auth_code=None, session_token=None):
         """
         Authenticates the user using the passed in credentials.
```

Comment:
Yep, adding the info... But this gives me a weird feeling... Like if I do have a session token, I do not need to call _authenticate()...  Reworking the code.